### ISSUE TITLE
[Feature] Make expect configurable when not using test runner

### ISSUE THREAD

**Issue by Georgegriff** [microsoft/playwright#28702]
We are starting to use artillery with its playwright engine https://github.com/artilleryio/artillery/tree/main/packages/artillery-engine-playwright.

Artillery has its own runner so it does not interact with playwright within a test. This means we cannot use the playwright.config.ts or use features that only work inside of a test.

We have lots of existing page object models with lots of use of `expect` from `@playwright/test` we want to re-use as much as possible, things are working really well except for we cannot configure:

```ts
{
expect: {
    timeout: 15000,
}
}
```

I know we can call `.configure` and create our own modified expect but this does not seem to work for locators.
```ts
export const expect = baseExpect.configure({
  timeout: 15000,
});
```
i tried this and it seems to have no effect
![image](https://github.com/microsoft/playwright/assets/9056958/88b6a2c9-cc74-4c35-8b64-c7bd66d3be8b)

I think this is happening here: https://github.com/microsoft/playwright/blob/e2c335a40cc49edbfc7a0b0596f71d4058f8dcfb/packages/playwright/src/matchers/expect.ts#L246 it seems there is no way to get the expect configuration into here.

I'm not sure if this is locator-specific or not.

**Comment by yury-s** [comment]
> it seems there is no way to get the expect configuration into here.
> I wonder if the early return could respect this setting by moving this code?

Yes, at first glance, it should be possible to move [setCurrentExpectConfigureTimeout](https://github.com/microsoft/playwright/blob/e2c335a40cc49edbfc7a0b0596f71d4058f8dcfb/packages/playwright/src/matchers/expect.ts#L286) before early return. Would you like to open a PR?

**Comment by Georgegriff** [comment]
I'll take a look tomorrow thanks!
